org.mwc.cmap.legacy/src/MWC/GUI/Layers.java

```diff
@@ -1410,6 +1410,9 @@ public boolean hasNoBounds(Plottable newItem) {
 				Plottable p = (Plottable)edits.nextElement();
 				if(p.getBounds()!=null) {
 					noBoundsCount++;
+					//the very first shape is already added at this stage,
+					//so we check for future additions to prevent rescaling every time
+					// a shape is added
 					if(noBoundsCount>1) {
 						hasNoBounds=false;
 						break;

```

Comment:
@ayeshare - I'm pretty sure this will only break out of the inner loop.  I think it would be fixed if you switch to `return false`, and have `return true` at the end of the method.